.goreleaser.yml error: field replacements not found in type config.Archive

# TL;DR;

`archives.replacements` has been deprecated and is now removed in recent `goreleaser` https://goreleaser.com/deprecations/#archivesreplacements and needs to be fixed in the `.goreleaser.yml`.

# Details

When running `make` I get this error:

```
$ make
goreleaser --snapshot --skip-publish --rm-dist
Flag --rm-dist has been deprecated, please use --clean instead
  • starting release...
  • loading config file                              file=.goreleaser.yml
  ⨯ release failed after 0s                  error=yaml: unmarshal errors:
  line 36: field replacements not found in type config.Archive
make: *** [Makefile:3: bin] Error 1
```

Checking the YAML with `goreleaser check` produces the same error:

```
$ goreleaser check
  • loading config file                              file=.goreleaser.yml
  ⨯ command failed                                   error=yaml: unmarshal errors:
  line 36: field replacements not found in type config.Archive
```

I figured I check the JSONSchema (`goreleaser jsonschema`) and indeed it looks like the `Archive` type has no property named `replacements`:

```json
{
	"Archive": {
				"properties": {
					"id": {
						"type": "string"
					},
					"builds": {
						"items": {
							"type": "string"
						},
						"type": "array"
					},
					"builds_info": {
						"$ref": "#/$defs/FileInfo"
					},
					"name_template": {
						"type": "string"
					},
					"format": {
						"type": "string",
						"enum": [
							"tar",
							"tgz",
							"tar.gz",
							"zip",
							"gz",
							"tar.xz",
							"txz",
							"binary"
						],
						"default": "tar.gz"
					},
					"format_overrides": {
						"items": {
							"$ref": "#/$defs/FormatOverride"
						},
						"type": "array"
					},
					"wrap_in_directory": {
						"oneOf": [
							{
								"type": "string"
							},
							{
								"type": "boolean"
							}
						]
					},
					"strip_parent_binary_folder": {
						"type": "boolean"
					},
					"rlcp": {
						"oneOf": [
							{
								"type": "string"
							},
							{
								"type": "boolean"
							}
						],
						"description": "you can now remove this"
					},
					"files": {
						"items": {
							"$ref": "#/$defs/File"
						},
						"type": "array"
					},
					"meta": {
						"type": "boolean"
					},
					"allow_different_binary_count": {
						"type": "boolean"
					}
				},
				"additionalProperties": false,
				"type": "object"
			}
}
```

The error only occurs since a recent goreleaser version but the `archives.replacements` field was deprecated before: https://goreleaser.com/deprecations/#archivesreplacements

Apparently others have the same problem: https://github.com/pulumi/pulumi/issues/13312.

--- a/.goreleaser.yml
+++ b/.goreleaser.yml
@@ -32,13 +32,14 @@ builds:
     ldflags:
       - -s -w -X main.version={{.Version}} -X main.commit={{.Commit}} -X main.date={{.CommitDate}} -X main.builtBy=gorelease
 archives:
-  - name_template: '{{ .ProjectName }}_{{ .Os }}_{{ .Arch }}{{ if .Arm }}v{{ .Arm }}{{ end }}'
-    replacements:
-      darwin: macos
-      linux: linux
-      windows: windows
-      386: i386
-      amd64: x86_64
+  - name_template: >-
+      {{ .ProjectName }}_
+      {{- if eq .Os "Darwin" }}macos_
+      {{- else }}{{- tolower .Os }}_{{end}}
+      {{- if eq .Arch "amd64" }}x86_64
+      {{- else if eq .Arch "386" }}i386
+      {{- else }}{{ .Arch }}
+      {{ if .Arm }}v{{ .Arm }}{{ end }}
     format_overrides:
     - goos: windows
       format: zip
@@ -55,7 +56,7 @@ changelog:
 brews:
   - 
     name: spr
-    tap:
+    repository:
       owner: ejoffe
       name: homebrew-tap
     homepage: https://github.com/ejoffe/spr

